pkg/router/util.go

```diff
@@ -32,3 +32,18 @@ func filterMetadata(meta map[string]string) map[string]string {
 	meta[toolkitReconcileKey] = toolkitReconcileValue
 	return meta
 }
+
+// mergeMeta merges two maps with the second one overwriting the first one
+func mergeMeta(m1 map[string]string, m2 map[string]string) map[string]string {
```

Comment:
If anyone knows a more Kubernetes native function that already exists I'm happy to use that. I just couldn't find any (even though I thought I used one in the past)